Jint/Native/Function/FunctionPrototype.cs

```diff
@@ -65,19 +65,13 @@ private static JsValue HasInstance(JsValue thisObj, JsValue[] arguments)
 
         private JsValue Bind(JsValue thisObj, JsValue[] arguments)
         {
-            if (!(thisObj is ICallable))
+            if (thisObj is not ICallable)
             {
                 ExceptionHelper.ThrowTypeError(Engine, "Bind must be called on a function");
             }
 
             var thisArg = arguments.At(0);
-            var f = new BindFunctionInstance(Engine)
-            {
-                TargetFunction = thisObj,
-                BoundThis = thisObj is ArrowFunctionInstance ? Undefined : thisArg,
```

Comment:
So no more ArrowFunctionInstance, less code repetition then?

Reply:
Yes, they now follow the same logic. We can separate again later if we want to fine-tune, but spec doesn't really talk about it being a separate concept, just configured differently.